test/new-e2e/tests/ndm/snmp/snmp_test.go

```diff
@@ -155,3 +164,37 @@ func (s *snmpDockerSuite) TestSnmp() {
 		assert.Contains(c, metrics, "snmp.sysUpTimeInstance", "metrics %v doesn't contain snmp.sysUpTimeInstance", metrics)
 	}, 5*time.Minute, 10*time.Second)
 }
+
+func (s *snmpDockerSuite) TestSnmpTagsAreStoredOnRestart() {
+	fakeintake := s.Env().FakeIntake.Client()
+	var initialMetrics []*aggregator.MetricSeries
+	var err error
+
+	require.EventuallyWithT(s.T(), func(t *assert.CollectT) {
+		initialMetrics, err = fakeintake.FilterMetrics("snmp.device.reachable")
+		assert.NoError(s.T(), err)
```

Comment:
This will make the whole `TestSnmpTagsAreStoredOnRestart` test fail instead of retrying inside the `EventuallyWithT` block.
We need to target the `CollectT` of the `EventuallyWithT` function.

```suggestion
		assert.NoError(t, err)
```

Reply:
changed https://github.com/DataDog/datadog-agent/pull/26003/commits/0cbe4239a32b07b1ae67d0ba54de0e013fa88658